src/main/java/uk/gov/pay/ledger/agreement/resource/AgreementResource.java

```diff
@@ -44,8 +49,31 @@ public AgreementSearchResponse search(@Valid @BeanParam AgreementSearchParams qu
     @GET
     @Timed
     public Agreement get(@PathParam("agreementExternalId") String agreementExternalId,
+                         @HeaderParam(HEADER_PARAM_X_CONSISTENT) Boolean isConsistent,
                          @Context UriInfo uriInfo) {
-        return agreementService.findAgreement(agreementExternalId)
-                .orElseThrow(() -> new WebApplicationException(Response.Status.NOT_FOUND));
+        if (isConsistent != null && isConsistent) {
+            try {
+                var eventDigest = eventService.getEventDigestForResource(agreementExternalId);
+
+                // compare the current event stream to the existing projection snapshot
+                return agreementService.findAgreementEntity(agreementExternalId)
+                        .map(agreementEntity -> {
+                            if (agreementEntity.getEventCount() < eventDigest.getEventCount()) {
+                                // the projection from the database is behind, rebuild a projection using the latest events
```

Comment:
Suggest we have a standard log pattern (maybe extract to an overriden private method) to show the following for every request:
- was consistent requested
- did we rebuild from the event stream or serve the projection

I'd expect these numbers will be useful for tuning this events pattern.